cmd/client-s3.go

```diff
@@ -1368,37 +1388,11 @@ func (c *S3Client) Stat(isIncomplete, isFetchMeta, isPreserve bool, sse encrypt.
 		if strings.HasSuffix(objectStat.Key, string(c.targetURL.Separator)) {
```

Comment:
Why are we omitting this check - `strings.Contains(objectStat.Key, object)` here

Reply:
> Why are we omitting this check - `strings.Contains(objectStat.Key, object)` here? Is it because we already do a stat check before?

Yes right. I wanted to avoid listing at all cost here since listing can take awhile if there are many objects